src/modules/CRUD/services/CRUDProvider.js

```diff
@@ -139,17 +140,24 @@ export class CRUDProvider extends React.Component {
       result.error = e;
     }
 
-    const { feature = {}, error = {} } = result;
+    const { feature: newFeature = {}, error = {} } = result;
 
-    this.setState(({ errors, featuresList }) => ({
-      featuresList: updateOrSaveFeatureInFeaturesList(featuresList, feature),
+    this.setState(({ errors, feature, featuresList }) => ({
+      feature: {
```

Comment:
And what about an pretty `updateFeatureIdentifier()` function? :smiley: 

like:
```suggestion
      feature: updateFeatureIdentifier(feature, newFeature),
```